### ISSUE TITLE
Transpilation issue with `this` remaining in function args

### ISSUE THREAD

**Issue by Techn1x** [ember-cli/ember-template-imports#223]
I have a GTS file like the following;

```ts
function someFunction(this: SomeContext, data: Row[]) {
  const filter = this.filter
  return data.map(...)
}

class extends Component<Signature> {
  get something() {
    someFunction.call(this.args.someOtherContext, this.rows)
  }

  <template>
    {{#each this.something as |row|}}
      ...
    {{/each}}
  </template>
}
```

It's a contrived example, but the point is there's the `someFunction` function in the GTS file that has `this: SomeContext` intended to be a type hint, to be transpiled away.

When debugging the code in browser, I can see that it is present as a real arg.
![Screenshot 2023-12-07 at 3 22 53 pm](https://github.com/ember-template-imports/ember-template-imports/assets/1049837/f27bae90-ba2f-47b1-8d4c-ac88173417bd)
(I also checked the source without sourcemaps, and can see `this1` is still present in the args)

_This means the actual named argument `data` becomes undefined in the function._

Some findings;
* This was not an issue in the original TS file - it was only once the file was converted to GTS
* The application does not have ember-cli-typescript installed - but installing it makes no difference to the output.

**Comment by backspace** [comment]
I’m seeing this too in [this PR to update to 4.1.1](https://github.com/cardstack/boxel/pull/1302). We have [tests that use an extended `this`](https://github.com/cardstack/boxel/blob/dee3d030cc4be4c13a4cefb3feb870b54dd29678/packages/boxel-ui/test-app/tests/integration/components/resizable-panel-group-test.gts#L138), they are unable to access the QUnit `assert` as it is `undefined`.

<img width="1139" alt="Boxel Tests 2024-05-28 14-32-57" src="https://github.com/ember-template-imports/ember-template-imports/assets/43280/15ba2112-52a3-4677-8faf-3419c48d89df">

It [shows up](https://github.com/cardstack/boxel/actions/runs/9275028938/job/25518773189?pr=1302#step:5:69) in tests as a failed call to a custom assertion.

**Comment by NullVoxPopuli** [comment]
Have you also updated:
- ember-cli-babel 
- babel-plugin-ember-template-compilation (should be at least 2.2.5 in your lockfile)
- @embroider/compat 
- @emmroider/core
- ember-auto-import 

?

V4 of ember-template-imports requires newer babel transpilation settings, provided by the above packages (embroider not required if you aren't using it, fwiw)

**Comment by backspace** [comment]
Thanks, should these be mentioned [here](https://github.com/ember-template-imports/ember-template-imports/blob/master/CHANGELOG.md#v400-2023-10-18)? I do see it [here](https://github.com/ember-template-imports/ember-template-imports/releases/tag/v4.0.0).

I updated each of these [in a series of commits](https://github.com/cardstack/boxel/pull/1302) (`babel-plugin-ember-template-compilation` seemed to need to happen with the Embroider updates) and sadly the problem remained.

However I looked for other tests that used a custom test context because I was sure we had them in `.gts` files and found that they were using a different style:

```
  test<TestContextWithSave>('can create new card-instance file in local realm with card type from same realm', async function (assert) {
```

I changed the tests to follow that pattern and the problem went away. (Some tests are still failing but seemingly not due to this problem.) Maybe this is more of a workaround than a true fix, but it’s worth trying?

**Comment by ef4** [comment]
This bug is covered by https://github.com/embroider-build/content-tag/issues/71

**Comment by NullVoxPopuli** [comment]
should be fixed now

**Comment by Techn1x** [comment]
This does seem to be fixed now, once on content-tag 2.0.3

Thanks a bunch!
